geonode/base/utils.py

```diff
@@ -55,23 +55,22 @@ def delete_orphaned_thumbs():
     """
     Deletes orphaned thumbnails.
     """
-    if isinstance(storage, FileSystemStorage):
-        documents_path = os.path.join(settings.MEDIA_ROOT, 'thumbs')
-    else:
-        documents_path = os.path.join(settings.STATIC_ROOT, 'thumbs')
-    if os.path.exists(documents_path):
-        for filename in os.listdir(documents_path):
-            fn = os.path.join(documents_path, filename)
-            model = filename.split('-')[0]
-            uuid = filename.replace(model, '').replace('-thumb.png', '')[1:]
-            if ResourceBase.objects.filter(uuid=uuid).count() == 0:
-                print('Removing orphan thumb %s' % fn)
-                logger.debug('Removing orphan thumb %s' % fn)
-                try:
-                    os.remove(fn)
-                except OSError:
-                    print('Could not delete file %s' % fn)
-                    logger.error('Could not delete file %s' % fn)
+    deleted = []
+    files = get_thumbs()
+
+    for filename in files:
+        model = filename.split('-')[0]
+        uuid = filename.replace(model, '').replace('-thumb.png', '')[1:]
+        if ResourceBase.objects.filter(uuid=uuid).count() == 0:
```

Comment:
This will make one DB request per file, which can significantly slow things down. Since you're already re-writing this method I suggest something like below:

```python
def get_uuid(filename):
    # You may want to think about error handling here, ie what if the
    # `filename` argument doesn't follow the convention you expect
    # (if that's a situation that your code can feasibly run into)?
    # Maybe regex is better to validate correct format and extract the bit you need?
    model = filename.split('-')[0]
    return filename.replace(model, '').replace('-thumb.png', '')[1:]

...
# Get a set of UUIDs (to ensure uniqueness) 
uuids = {get_uuid(name) for name in get_thumbs()}

# Match uuids against the DB in a single query. This is likely better than executing multiple queries,
# however note that a) depending on which UUID version is used such DB  fields can be slow to look up,
# b) if the list of uuids is very long the `IN` queries constructed by Django may not be the fastest.
db_uuids = ResourceBase.objects.filter(uuid__in=uuids).values_list('uuid', flat=True)

# Iterate over all orphaned uuids (without a DB match):
for uuid in uuids - set(db_uuids):
    try:
        remove_thumb(uuid)
        deleted.append(uuid)
    except NotImplementedError as e:
        logger.error("Failed to delete orphaned thumbnail '{}': {}".format(filename, e))

return deleted
```

You could go even further by making your `remove_thumb` handle & log errors internally and return either the removed file's uuid or `None` on failure. In that case the last few lines of this method would like below. However, I'm not sure whether that's worth the extra effort.

```python
orphaned_uuids = uuids - set(db_uuids)
deleted = (remove_thumb(uuid) for uuid in orphaned_uuids)
return [uuid for uuid in deleted if uuid is not None]
```